File: Serene/Serene.Script/Imports/FormContexts/Administration.LanguageForm.cs

namespace Serene.Administration
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Runtime.CompilerServices;

    public partial class LanguageForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Administration.Language";
    
        public LanguageForm(string idPrefix) : base(idPrefix) {}
    
        public StringEditor LanguageId { [InlineCode("{this}.w('LanguageId', Serenity.StringEditor)")] get; private set; }
        public StringEditor LanguageName { [InlineCode("{this}.w('LanguageName', Serenity.StringEditor)")] get; private set; }
    }
}

File: Serene/Serene.Web/Imports/ServerTypings/Administration.LanguageForm.ts
namespace Serene.Administration {
    export interface LanguageForm {
        LanguageId: Serenity.StringEditor;
        LanguageName: Serenity.StringEditor;
    }

    export class LanguageForm extends Serenity.PrefixedContext {
        static formKey = 'Administration.Language';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!LanguageForm.init)  {
                LanguageForm.init = true;

                var s = Serenity;
                var w0 = s.StringEditor;

                Q.initFormType(LanguageForm, [
                    'LanguageId', w0,
                    'LanguageName', w0
                ]);
            }
        }
    }
}

